genomedata/_chromosome.py

```diff
@@ -0,0 +1,689 @@
+        return self._name
+
+    @property
+    def mode(self):
+        """Return the read/write properties for this chromosome as string"""
+        return Chromosome.default_mode
+
+    @property
+    def num_tracks_continuous(self):
+        """Return the number of tracks in this chromosome"""
+        try:
+            return len(self.tracknames_continuous)
+        except AttributeError:
+            return 0
+
+    @property
+    def seq(self):
+        """Return the genomic sequence of this chromosome.
+
+        If the index or slice spans a non-supercontig range, N's are
+        inserted in place of the missing data and a warning is issued.
+
+        Example:
+
+        >>> chromosome = genome["chr1"]
+        >>> for supercontig in chromosome:
+        ...     print repr(supercontig)
+        ...
+        <Supercontig 'supercontig_0', [0:121186957]>
+        <Supercontig 'supercontig_1', [141476957:143422081]>
+        <Supercontig 'supercontig_2', [143522081:247249719]>
+        >>> chromosome.seq[0:10].tostring()  # Inside supercontig
+        'taaccctaac'
+        >>> chromosome.seq[121186950:121186970].tostring()  \
+# supercontig boundary
+        'agAATTCNNNNNNNNNNNNN'
+        >>> chromosome.seq[121186957:121186960].tostring()  \
+# not in supercontig
+        OverlapWarning: slice of chromosome sequence does not overlap any \
+supercontig (filling with 'N')
+        'NNN'
+
+        The entire sequence for a chromosome can be retrieved with:
+
+        >>> chromosome.seq[chromosome.start:chromosome.end]
+
+        """
+        return self._seq
+
+    @property
+    def supercontigs(self):
+        """Return the supercontig that contains this range if possible.
+
+        :returns: :class:`Supercontig`
+
+        Indexable with a slice or simple index:
+
+        >>> chromosome.supercontigs[100]
+        [<Supercontig 'supercontig_0', [0:66115833]>]
+        >>> chromosome.supercontigs[1:100000000]
+        [<Supercontig 'supercontig_0', [0:66115833]>, \
+<Supercontig 'supercontig_1', [66375833:90587544]>, \
+<Supercontig 'supercontig_2', [94987544:199501827]>]
+        >>> chromosome.supercontigs[66115833:66375833]  \
+# Between two supercontigs
+        []
+
+        """
+        return self._supercontigs
+
+
+class _ChromosomeList(object):
+    def __init__(self):
+        self.open_chromosomes = {}  # NB: maintained by subclass
+
+    def __getitem__(self, name):
+        """Returns an existing opened chromosome if it exists. Subclasses
+        assumed to implement this method and call this for caching."""
+        # If we have opened the chromosome already
+        if (name in self.open_chromosomes and
+           self.open_chromosomes[name].isopen):  # NB: manual closure check
+            # Return cached object
+            return self.open_chromosomes[name]
+
+        return None
+
+    def close(self):
+        self.open_chromosomes = {}
+
+
+class Supercontig(object):
```

Comment:
is the plan for removal of dependence on this to HDF5 a future PR?

Reply:
It will be in this same PR